File: backend/zvms/views/volunteers.py
from zvms.routelib import *
from zvms.res import *
from zvms.views.structs import Volunteer
import zvms.impls.volunteers

route(
    rule='/volunteers',
    method='GET',
    impl_func=zvms.impls.volunteers.search_volunteers
)

route(
    rule='/volunteers/<int:id>',
    method='GET',
    impl_func=zvms.impls.volunteers.get_volunteer_info,
)

route(
    rule='/volunteers',
    method='POST',
    impl_func=zvms.impls.volunteers.create_volunteer,
    params=Volunteer,
    auth=AUTH.MANAGER | AUTH.TEACHER
)

route(
    rule='/volunteers/<int:id>',
    method='PUT',
    impl_func=zvms.impls.volunteers.update_volunteer,
    params=Volunteer,
    auth=AUTH.MANAGER | AUTH.TEACHER
)

route(
    rule='/volunteers/<int:id>',
    method='DELETE',
    impl_func=zvms.impls.volunteers.delete_volunteer,
    auth=AUTH.MANAGER | AUTH.TEACHER
)

File: apis/paths/volunteers.ts
import { $ } from "zvms-apis-paths-gen";
import { authData as a } from "../users/index.js";
import { str, structs } from "../types/index.js";

export default $(
    "/volunteers",
    [],
    {
        type: "GET",
        name: "search_volunteers"
    },
    {
        type: "POST",
        name: "create_volunteer",
        auths: [a.createVol],
        req: structs.VolunteerRecord
    },
    $(
        "/<int:id>",
        [],
        {
            type: "GET",
            name: "get_volunteer_info",
            desc: ""
        },
        {
            type: "PUT",
            name: "update_volunteer",
            auths: [a.updateVol],
            req: structs.VolunteerRecord
        },
        {
            type: "DELETE",
            name: "delete_volunteer",
            auths: [a.deleteVol]
        }
    )
);